### ISSUE TITLE
Feature Request: `str_replace()`, `str_replace_all()` like renaming within `cols_label()` to apply across as many columns as applicable

### ISSUE THREAD

**Issue by ScoobyQ** [rstudio/gt#701]
I would like to see cols_label support multi-column renames using pattern matching and specified replacement strings for matches much like str_replace, str_replace_all.

If I had columns such as: `c1_mean_2019-20`, `c1_mean_2020-21`, I would like to be able to replace `c1_mean_` with `''`, leaving  `2019-20`, `2020-21`. This needs to happen **IN** the gt object.

Something like:

```r
cols_label( contains('c1_mean_') = str_replace(contains('c1_mean_'),'c1_mean_', '') )
```

Looking at [cols_label](https://gt.rstudio.com/reference/cols_label.html) this functionality does not appear to exist.

**Comment by rich-iannone** [comment]
This is a good idea! Could work in the form of

```r
cols_label(contains("...") = ~ fn(.))
```

So that any string manipulation functions would work. We'd have to also ensure that helpers such as `md()`, `html()`, etc., continue to work in this form

```r
cols_label(contains("...") = md(~ fn(.)))
```

and this

```r
cols_label(contains("...") = ~ md(fn(.)))
```

**Comment by rich-iannone** [comment]
Fixed by #1158.
